pil-analyzer/src/condenser.rs

```diff
@@ -309,13 +310,37 @@ impl<'a, T: FieldElement> SymbolLookup<'a, T> for Condenser<'a, T> {
         Ok(Value::Integer(degree.into()).into())
     }
 
-    fn new_witness_column(
+    fn new_column(
         &mut self,
         name: &str,
+        value: Option<Arc<Value<'a, T>>>,
         source: SourceRef,
     ) -> Result<Arc<Value<'a, T>>, EvalError> {
         let name = self.find_unused_name(name);
-        let kind = SymbolKind::Poly(PolynomialType::Committed);
+        let kind = SymbolKind::Poly(if value.is_some() {
+            PolynomialType::Constant
+        } else {
+            PolynomialType::Committed
+        });
+        let value = value.map(|v|{
+            if let Value::Closure(evaluator::Closure {
+                lambda,
+                environment: _,
+                type_args: _,
+            }) = v.as_ref()
+            {
+                if !lambda.outer_var_references.is_empty() {
+                    return Err(EvalError::TypeError(format!("Lambda expression for fixed column {name} must not reference outer variables.")))
+                }
+                Ok(FunctionValueDefinition::Expression(TypedExpression {
+                    e: Expression::LambdaExpression(source.clone(), (*lambda).clone()),
+                    type_scheme: None,
+                }))
+            } else {
+                Err(EvalError::TypeError(format!("Only lambda expressions are allowed for dynamically-created fixed columns. Got {v}.")))
```

Comment:
Is there a test that reaches this?

Reply:
I don't think so, it should not be able to pass the type checker.

Reply:
Added a test.

Reply:
Should I change this here to panic?

Reply:
Err sounds better